modules/ROOT/pages/availability_scaling/index.adoc

```diff
@@ -0,0 +1,140 @@
++
+For availability, storing user files is by nature mandatory to be on xref:prerequisites/index.html#filesystems-and-external-storage[shared storage] to be accessible by the nodes and/or services.
+
+==== Classic High Availability
+
+When using the classic form of high availability (HA), you can either create a setup where the nodes are both active (active-active) or one node only serves as a fallback, waiting for a failure to occur (active-passive). With an A-A setup, a load balancer is needed in front of the nodes. In case of a failure, the remaining node has to be capable of taking over all the load they shared before the other node failed.
+
+image::availability_scaling/a_a_ha_with_data_server.svg[width=310]
+
+Summary::
+This use case can be considered if hardware availability is the primary objective. When using an active-active configuration, each node is addressed by a load balancer for load distribution. This requires that the nodes have the same setup and services are bound to the nodes. *Scalability is hardware dependent and in case of an active-passive setup, you can even get reduced scalability*.
+
+==== Clustering
+
+The main objective for clustering is not only availability but also distributing load across _multiple_ nodes. With clustering a load balancer (LB) is mandatory.
+
+image::availability_scaling/a_a_cluster_with_data_server.svg[width=450]
+
+Summary::
+Clustering provides better availability and scalability for growing loads and covers fail-over if a node fails but it still focuses only on hardware. A cluster environment can grow very complex with many dependencies, see the section xref:availability[Availability]. *Scalability is much better but lacks when it comes to load-based dynamic assignment of services*.
+
+== Scalability
+
+There are two classic ways to achieve scalability, which is scale-up and scale-out.
+
+[width="100%",cols="^100%,^100%",frame=none,grid=none]
+|===
+2+| The pictures below show the different ways of classic scaling
+
+a| image::availability_scaling/scale_up.svg[width=214]
+a| image::availability_scaling/scale_out.svg[width=300]
+|===
+
+What may sound simple with regard to a _service_, services can be a complex topic in reality as they may contain a lot of software components and their configuration building it. Adding more services or migrating a service can therefore be a challenging task adding complexity and can introduce sources of error.
+
+Summary::
+Scalability and availability are often aligned to each other and the decision how to achieve the goal set can be a complex task. This becomes even more true when dynamic load balancing comes into play. *Because services consist of many components to take care of, real dynamic adoption and dynamic migration is hard to be achieved*.
```

Comment:
```suggestion
Scalability and availability are often aligned to each other and the decision how to achieve the goal set can be a complex task. This becomes even more true when dynamic load balancing comes into play. *Because services consist of many components to take care of, real dynamic adoption and dynamic migration is hard to achieve*.
```